FEAT : 로그인시 response 값에 companyId를 추가한다.

deehyeon opened the pull request:
## 💡 관련 이슈
- #61 


## 📝 작업 내용
### **1. 주요 변경 사항**
로그인 응답 DTO(MemberLoginResponse)에 companyId 필드 추가
- 개인 회원: companyId = null
- 기업 회원: companyId에 해당 회원이 속한 기업 ID

### **2. Response dto 단일화**
- 개인·기업 모두 동일한 Member 엔티티를 사용
- companyId는 기업 회원에게만 부가되는 Optional 정보이므로, null 혹은 빈 Optional로 표현
=> 응답 구조 일관성 및 매퍼·문서화 중복 방지를 위해 별도 DTO 분리 없이 하나의 응답으로 통일

## 🧑‍💻 변경 사항
> 변경된 사항을 알려주세요


## 💬 리뷰 요구사항(선택)
> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요
>
> ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요?


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* 신규 기능
  * 로그인 응답에 기업 회원의 companyId를 포함(개인/기업 분기 응답).
  * 기업 회원 가입 시 이메일 중복 검사 추가.

* API 변경
  * 기업 목록/상세 조회 응답을 상세 DTO로 통일해 더 풍부한 정보 제공.
  * 기업 회원 로그인 시 companyId 누락 시 422 오류 반환.

* 문서화
  * 주소 및 기업 등록 요청 필드에 Swagger 예시/설명 추가로 API 명세 개선.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

deehyeon committed:
#61 Feat : 로그인시 response 값에 companyId를 추가한다.

SeongHo5356 commented on src/main/java/hanieum/conik/domain/member/dto/MemberLoginResponse.java:
```diff
@@ -2,10 +2,15 @@
+        MemberInfo memberInfo,
+
+        @Schema(description = "멤버가 속한 기업 id", nullable = true)
+        Optional<Long> companyId
```
from 이나 of 등의 정적 팩토리 메서드 를 통해서 생성해주는 로직을 만들고 이걸 AuthService에서 사용할 수도 있을 것 같아요

deehyeon replied:
@SeongHo5356 
```java
public record MemberLoginResponse(
    TokenInfo      tokenInfo,
    MemberInfo     memberInfo,
    Optional<Long> companyId
) {
    public static MemberLoginResponse of(Member member, TokenInfo tokenInfo) {
        Optional<Long> companyId = member.getRole() == MemberRole.INDIVIDUAL
            ? Optional.empty()
            : Optional.of(member.getCompanyId());
        return new MemberLoginResponse(tokenInfo, MemberInfo.from(member), companyId);
    }
}
```
로그인 로직에서 이렇게 메서드를 만들어 사용할까 고민하긴 했었는데 이렇게 되면 ROLE에 따른 companyId의 반환값이 정해지는 비즈니스 로직이 record에 들어가게 된다고 생각해서 AuthService에 로직을 빼두었습니다. 
dto는 단순 데이터와 매핑만 책임지게 하려고 했는데 뭐가 더 좋은 방법인지는 잘 모루겠숴요..

SeongHo5356 replied:
그러면 파라미터로 Member member, TokenInfo tokenInfo를 갖는 of와
파라미터로 Member member, TokenInfo tokenInfo, Long CompanyId를 갖는 of 이렇게 두개를 선언해두고
불러만 쓰는 건 어떤가요?

deehyeon replied:
@SeongHo5356 일반 멤버인 경우랑 기업 멤버인 경우를 확실하게 구분해서 표현하고 싶어서 그런 걸까요??

deehyeon replied:
```java
public static MemberLoginResponse of(TokenInfo tokenInfo, MemberInfo memberInfo, Optional<Long> companyId) {
                return new MemberLoginResponse(tokenInfo, memberInfo, companyId.isEmpty() ? Optional.empty() : companyId);
        }
```
따로 두지 않고 이렇게 구현하는건 어떤가요??

SeongHo5356 replied:
Optional은 “주로 메서드의 반환 타입으로 사용하도록 의도됐다”고 JavaDoc에 명시돼 있다고 하네요. 따라서, 필드나  파라미터 타입으로 쓰라기 보단 결과가 없을 수 있는 상황에서 null 대신 쓰는 용도로 알맞을 것 같아요. 아래처럼 구현하는 게 좋을 것 같네요!

```suggestion
{
    // 1개로 처리
    public static MemberLoginResponse of(TokenInfo token, Member member) {
        Long cid = (member.getRole() == MemberRole.INDIVIDUAL) ? null : member.getCompanyId();
        return new MemberLoginResponse(token, MemberInfo.from(member), cid);
    }

    // 역할 구분
    public static MemberLoginResponse individual(TokenInfo token, Member member) {
        return new MemberLoginResponse(token, MemberInfo.from(member), null);
    }
    public static MemberLoginResponse corporate(TokenInfo token, Member member, Long companyId) {
        return new MemberLoginResponse(token, MemberInfo.from(member), Objects.requireNonNull(companyId));
    }
}
```

SeongHo5356 commented on src/main/java/hanieum/conik/application/member/AuthService.java:
```diff
@@ -45,7 +48,8 @@ public MemberLoginResponse signUpIndividual(MemberSignUpRequest request) {
     @Override
     public MemberLoginResponse signUpCompanyMember(MemberSignUpRequest request, Long companyId) {
-        Company company = companyRepository.findById(companyId).orElseThrow(() -> new CompanyException(CompanyErrorType.COMPANY_NOT_FOUND));
+        Company company = companyRepository.findById(companyId)
+                .orElseThrow(() -> new CompanyException(CompanyErrorType.COMPANY_NOT_FOUND));
```
추후에 회사 찾는게 견적서 작성하거나, 회사 마이페이지 가거나 할때 많이 쓰일 수도 있을 것 같아서 인터페이스 만들어서 구현해주면 좋을 것 같아요

SeongHo5356 commented on src/main/java/hanieum/conik/application/member/AuthService.java:
```diff
@@ -61,17 +65,15 @@ public MemberLoginResponse login(MemberLoginRequest request) {
+
+        Optional<Long> companyId = getCompanyId(member);
+
+        return new MemberLoginResponse(tokenInfo, MemberInfo.from(member), companyId);
```
같은 리뷰인데 해당 MemberLoginResponse의 정적 팩토리 메서드를 통해서 생성해줄 수도 있을 것 같네요.
추가로, getTokenInfo(member) 나 getCompanyId(member)를 바로 인자로 줄 수도 있었을 것 같은데 변수로 선언한 건 가독성 때문인가요?

deehyeon replied:
@SeongHo5356  넵 맞아요
로그인 로직이 어떤 흐름으로 이어지는지(비밀번호 검증 -> 토큰 생성 -> companyId 찾기) 흐름을 보여주고싶어서 이렇게 했는데 바로 인자로 주는게 더 깔끔할까요??

SeongHo5356 replied:
아하 뭔가 그런 고민보다는, 저는 사용하지 않는 변수를 선언하는 거에 대해서 비효율적이라고 생각해서 했던 리뷰였는데, 비즈니스 흐름을 보여주고 싶다면 이것도 좋을 것 같아요

SeongHo5356 committed:
#65 Refactor: 회사 세부 정보를 조회한다

# Conflicts:
#	src/main/java/hanieum/conik/application/company/CompanyService.java
#	src/main/java/hanieum/conik/application/company/required/CompanyRepository.java

deehyeon committed:
#62 Feat : MemberFinder에 findByEmail 추가

# Conflicts:
#	src/main/java/hanieum/conik/application/proposal/ProposalModifyService.java
#	src/main/java/hanieum/conik/application/proposal/ProposalQueryService.java

deehyeon committed:
#62 Feat : dto에서 Optional 타입 제거

deehyeon committed:
#62 Feat : 회사 등록 시 이메일 인증 로직 추가

deehyeon committed:
#62 Feat : 회사 등록 시 이메일 인증 로직 추가

deehyeon committed:
#62 Feat : 회사 등록 시 이메일 인증 로직 추가

deehyeon committed:
#62 Feat : 기업 등록 dto에 Schema example 추가

deehyeon committed:
#62 Feat : Error 타입 수정

deehyeon committed:
#62 Feat : Error 타입 수정

deehyeon committed:
#62 Feat : 회원가입 유형에 따른 response dto 정적 팩토리 메서드 분기

deehyeon committed:
#61 Feat : MemberFinder에 findByEmail을 추가한다.

deehyeon committed:
#62 Fix : 전체 기업조회시 기업의 주소를 제외한 나머지 기업 조회를 반환한다.

deehyeon committed:
#61 Fix : MemberFinder에서 find 메서드명을 findById로 수정한다.

EunjinWoo reviewed (approved):
고생했어어,,, 😭😭😭😭😭
